keep_args_alive error

### Describe the bug

When commenting in `@[keep_args_alive]` in [this code](https://github.com/antono2/tmp/blob/bff89170b7de949bdb6d49015156940aa754d6f1/main.v#L75)
and compiling with `v -show-c-output run .`
```
/tmp/v_1000/tmp2.01JMWJJB72VG0CG1TK9VFANR2K.tmp.c: In function ‘main__get_stub_main’:
/tmp/v_1000/tmp2.01JMWJJB72VG0CG1TK9VFANR2K.tmp.c:6031:58: error: invalid initializer
 6031 |         Array_fixed_main__Flag_bits_main_2 __tmp_arg_1 = param;
```

### Reproduction Steps

```
git clone -b keep_args_alive https://github.com/antono2/tmp
cd tmp
v run .
# comment in keep_args_alive
v -show-c-output .
```

### Expected Behavior

Should not stop compiling if V syntax is correct.

### Current Behavior

```
/tmp/v_1000/tmp2.01JMWJJB72VG0CG1TK9VFANR2K.tmp.c: In function ‘main__get_stub_main’:
/tmp/v_1000/tmp2.01JMWJJB72VG0CG1TK9VFANR2K.tmp.c:6031:58: error: invalid initializer
 6031 |         Array_fixed_main__Flag_bits_main_2 __tmp_arg_1 = param;
```

### Possible Solution

_No response_

### Additional Information/Context

_No response_

### V version

V 0.4.9 f3d2eb1

### Environment details (OS name and version, etc.)

|V full version      |V 0.4.9 9921692.f3d2eb1
|:-------------------|:-------------------
|OS                  |linux, Ubuntu 24.04.2 LTS
|Processor           |8 cpus, 64bit, little endian, Intel(R) Core(TM) i7-6700K CPU @ 4.00GHz
|Memory              |0.54GB/15.56GB
|                    |
|V executable        |/home/anton/workspace/v/v
|V last modified time|2025-02-24 18:11:14
|                    |
|V home dir          |OK, value: /home/anton/workspace/v
|VMODULES            |OK, value: /home/anton/.vmodules
|VTMP                |OK, value: /tmp/v_1000
|Current working dir |OK, value: /home/anton/workspace/tmp
|                    |
|Git version         |git version 2.43.0
|V git status        |weekly.2024.41-974-gf3d2eb1c
|.git/config present |true
|                    |
|cc version          |cc (Ubuntu 13.3.0-6ubuntu2~24.04) 13.3.0
|gcc version         |gcc (GCC) 13.0.1 20230224 (experimental)
|clang version       |clang version 21.0.0git (https://github.com/llvm/llvm-project.git bd034ab1113cd3705fedf75d870dc4b7388d8f70)
|tcc version         |tcc version 0.9.28rc 2024-07-31 HEAD@1cee0908 (x86_64 Linux)
|tcc git status      |thirdparty-linux-amd64 0134e9b9-dirty
|emcc version        |N/A
|glibc version       |ldd (Ubuntu GLIBC 2.39-0ubuntu8.4) 2.39
> [!NOTE]
> You can use the 👍 reaction to increase the issue's priority for developers.
>
> Please note that only the 👍 reaction to the issue itself counts as a vote.
> Other reactions and those to comments will not be taken into account.

--- a/vlib/v/gen/c/fn.v
+++ b/vlib/v/gen/c/fn.v
@@ -2579,9 +2579,17 @@ fn (mut g Gen) keep_alive_call_pregen(node ast.CallExpr) int {
 		// save all arguments in temp vars (not only pointers) to make sure the
 		// evaluation order is preserved
 		expected_type := node.expected_arg_types[i]
-		typ := g.table.sym(expected_type).cname
-		g.write('${typ} __tmp_arg_${tmp_cnt_save + i} = ')
-		g.ref_or_deref_arg(arg, expected_type, node.language, false)
+		typ_sym := g.table.sym(expected_type)
+		typ := typ_sym.cname
+		if typ_sym.kind != .array_fixed {
+			g.write('${typ} __tmp_arg_${tmp_cnt_save + i} = ')
+			g.ref_or_deref_arg(arg, expected_type, node.language, false)
+		} else {
+			g.write('${typ} __tmp_arg_${tmp_cnt_save + i} = {0};')
+			g.write('memcpy(&__tmp_arg_${tmp_cnt_save + i}, ')
+			g.ref_or_deref_arg(arg, expected_type, node.language, false)
+			g.writeln(', sizeof(${typ}));')
+		}
 		g.writeln(';')
 	}
 	g.empty_line = false

--- a/vlib/v/gen/c/fn.v
+++ b/vlib/v/gen/c/fn.v
@@ -2585,7 +2585,7 @@ fn (mut g Gen) keep_alive_call_pregen(node ast.CallExpr) int {
 			g.write('${typ} __tmp_arg_${tmp_cnt_save + i} = ')
 			g.ref_or_deref_arg(arg, expected_type, node.language, false)
 		} else {
-			g.write('${typ} __tmp_arg_${tmp_cnt_save + i} = {0};')
+			g.writeln('${typ} __tmp_arg_${tmp_cnt_save + i} = {0};')
 			g.write('memcpy(&__tmp_arg_${tmp_cnt_save + i}, ')
 			g.ref_or_deref_arg(arg, expected_type, node.language, false)
 			g.writeln(', sizeof(${typ}));')

--- a/vlib/v/tests/c_function/code.c
+++ b/vlib/v/tests/c_function/code.c
@@ -1,0 +1,3 @@
+void foo(int arg[1]) {
+
+}

--- a/vlib/v/tests/c_function/code_test.c.v
+++ b/vlib/v/tests/c_function/code_test.c.v
@@ -1,0 +1,10 @@
+module main
+
+#include "@VMODROOT/code.c"
+
+@[keep_args_alive]
+fn C.foo(arg [1]int)
+
+fn test_main() {
+	C.foo([1]!)
+}

--- a/vlib/v/tests/c_function/v.mod
+++ b/vlib/v/tests/c_function/v.mod
@@ -1,0 +1,7 @@
+Module {
+	name: 'c_function'
+	description: ''
+	version: ''
+	license: ''
+	dependencies: []
+}

